src/Components/Server/src/ComponentHub.cs

```diff
@@ -110,6 +110,18 @@ public string StartCircuit(string baseUri, string uri)
                 return null;
             }
 
+            if (baseUri == null ||
+                uri == null ||
+                !Uri.IsWellFormedUriString(baseUri, UriKind.Absolute) ||
+                !Uri.IsWellFormedUriString(uri, UriKind.Absolute))
+            {
+                // We do some really minimal validation here to prevent obviously wrong data from getting in
+                // without duplicating too much logic.
+                Log.InvalidInputData(_logger);
+                NotifyClientError(Clients.Caller, $"The uris provided are invalid.");
```

Comment:
Can we change this to instead of returning a string, we return some result object and deliver the failure synchronously to the client?

Otherwise the client continues doing its thing, and can cause further errors that hide the original issue.

Reply:
That could be done but I'm not sure that it helps in any tangible way. Our client code already validates that we return a non-null circuit Id - so we effectively already return success/failure.

A misbehavior client can just continue to do whatever it wants, which is the case regardless of whether we return a result type.

Reply:
The change that makes sense here it to terminate the connection when a failure occurs, which is a separate PR.